### PR TITLE
Expose configuration for local validation of access tokens

### PR DESCRIPTION
**Pull request by robertjd**

### CODE REVIEW

File: lib/config.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ web:
         ttl: 3600
     password:
       enabled: true
+      validationStrategy: stormpath
```
Comment (rdegges) [inline lib/config.yml]: We should quote this, no?

**Comment by rdegges** [comment]
Also: it'd be cool to have an additional strategy: random-validation. Where we do math.random() and randomly either validate locally, or against the API. This way you get some tradeoff between performance and security.

File: lib/config.yml (hunk 1 above)
Comment (robertjd) [inline lib/config.yml]: For simple strings you don't have to use quotes in YAML, but for consistency I'll add quotes

**Comment by robertjd** [comment]
The idea of random validation is interesting, but I don't think should be part of this library.  That could be a plugin/module.

File: docs/authentication.rst [hunk 2]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
     });
 
 Behind the scenes we are issuing a OAuth2 Access Token and Refresh token for
```
Comment (typerandom) [inline docs/authentication.rst]: `Refresh token` should be `Refresh Token`.

File: docs/authentication.rst [hunk 3]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
-the user, and storing them in secure, HTTPS-Only cookies.  The maximum
-lifetime of the cookies is controlled by the expiration time of the Refresh
-Token.
+the user, and storing them in secure, HTTPS-Only cookies.  After the user has
+logged in, these cookies will be supplied on every request.  Our library will
+assert that the access token is valid.  If the access token is expired, we will
```
Comment (typerandom) [inline docs/authentication.rst]: For consistency `access token` should be `Access Token`.

File: docs/authentication.rst [hunk 4]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+attempt to refresh it with the refresh token.
```
Comment (typerandom) [inline docs/authentication.rst]: `refresh token` should be `Refresh Token`.

File: docs/authentication.rst [hunk 5]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+    Express-Stormpath's OAuth2 cookie feature will not interfere with any
+    existing cookie-based session middleware you might have.  The cookies that
+    Stormpath creates are used exclusively for Stormpath's purposes, so it's
+    safe to create your own separate sessions if needed.
+
+
+Setting Token Expiration Time
+.............................
+
+If you need to change the expiration time of the access token or refresh Token,
```
Comment (typerandom) [inline docs/authentication.rst]: `refresh Token` should be `Refresh Token`.

File: docs/authentication.rst [hunk 6]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+your Stormpath Application.  There you will find the settings for each token.
+
+Token Validation Strategy
+.........................
+
+When a request comes into your server, this library will use the access token
+and refresh token cookies to make an authentication decision.  The default
+validation strategy works like this:
+
+- Validate the signature and expiration time of the access token.  If the access
```
Comment (typerandom) [inline docs/authentication.rst]: `access token` should be `Access Token`.

File: docs/authentication.rst [hunk 7]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+  token is expired, attempt to get a new one by using the refresh token.
```
Comment (typerandom) [inline docs/authentication.rst]: `refresh token` should be `Refresh Token`.

File: docs/authentication.rst [hunk 8]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+
+- If the access token is expired and cannot be refreshed, deny the request
```
Comment (typerandom) [inline docs/authentication.rst]: `access token` should be `Access Token`.

File: docs/authentication.rst [hunk 9]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+
+- If the access token is not expired and the signature is valid, the library
```
Comment (typerandom) [inline docs/authentication.rst]: `access token` should be `Access Token`.

File: docs/authentication.rst [hunk 10]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+  makes a request to the Stormpath API to assert that the access token has not
```
Comment (typerandom) [inline docs/authentication.rst]: `access token` should be `Access Token`.

File: docs/authentication.rst [hunk 11]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+.. warning::
+
+  When using local validation, your server will not be aware of token revocation
+  or any changes to the associated Stormpath account.  **This is a security
+  risk.**
+
+  There are two suggested strategies for dealing with this risk:
+
+  * Use a short expiration time for your access tokens (such as one hour or
+    less).  This will limit the amount of time that the access token can be used
```
Comment (typerandom) [inline docs/authentication.rst]: `access token` should be `Access Token`.

File: docs/authentication.rst [hunk 12]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+    for validation.  Our library *always* makes a request to the Stormpath API when
+    we attempt to refresh an access token, so the refresh attempt will fail
```
Comment (typerandom) [inline docs/authentication.rst]: `access token` and `refresh token` should be `Access Token` and `Refresh Token`.

File: docs/authentication.rst [hunk 13]
```diff
@@ -37,19 +37,82 @@ To use cookie authentication, simply use the ``loginRequired`` middleware::
...
+    at this time if the refresh token has been revoked.
```
Comment (typerandom) [inline docs/authentication.rst]: `refresh token` should be `Refresh Token`.

**Comment by typerandom** [comment]
Except the small doc edits, I think this looks good! :+1: :)

**Pull request merged**